### ISSUE TITLE
Generate typespecs

### ISSUE THREAD

**Issue by sambooo** [obmarg/kazan#44]
I reckon it'd be nice to be able to check the validity of struct-modifying code with Dialyzer. Given the types are already present in the docs it seems the required information is there to do this.

**Comment by obmarg** [comment]
Yeah, I agree - it would be great to generate typespecs along with the structs/functions.

**Comment by obmarg** [comment]
Started work on this feature last night on [this branch](https://github.com/obmarg/kazan/tree/feature/generate-typespecs).  Proving to be a lot easier than I expected.
